cmd/kops/update_cluster.go

```diff
@@ -109,6 +110,7 @@ func NewCmdUpdateCluster(f *util.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().StringVar(&options.OutDir, "out", options.OutDir, "Path to write any local output")
 	cmd.Flags().BoolVar(&options.CreateKubecfg, "create-kube-config", options.CreateKubecfg, "Will control automatically creating the kube config file on your local filesystem")
 	cmd.Flags().StringVar(&options.Phase, "phase", options.Phase, "Subset of tasks to run: "+strings.Join(cloudup.Phases.List(), ", "))
+	cmd.Flags().StringSliceVar(&options.PhaseOverrides, "phase-overrides", options.PhaseOverrides, "comma separated list of phase overrides, example: assets=ExistsAndWarnIfChanges,network=ExistsAndValidates")
```

Comment:
I think it might be cleaner to have one flag here.  When we get down to the per-task overrides, what do you imagine the syntax to look like (e.g. `cluster,SecurityGroupRule=ExistsAndWarnIfChanges`)?

Reply:
Personally I would rather have this in the api, but another PR for that.

For one I strongly disagree with you. It is super confusing to me to not have a separate flag.

I understand where you are coming from, and I can see you point, but I would typo a single flags so easily.

Reply:
> I think it might be cleaner to have one flag here. When we get down to the per-task overrides, what do you imagine the syntax to look like (e.g. cluster,SecurityGroupRule=ExistsAndWarnIfChanges)?

Need more context here.  Are you recommending re-using the phases flag?

Reply:
I'm asking how we should express: "cluster phase, but do ExistsAndWarnIfChanges on the SecurityGroupRules".  Or - for example - "iam, but skip security groups" or "only security groups".  If we get that right, we don't need #4434 for every single special-case.

Reply:
So here is an example of what this PR provides:

```
kops update cluster --phase cluster --phase-overrides "assets=Ignore,network=ExistsAndValidates,iam= WarnIfInsufficientAccess,security= ExistsAndWarnIfChanges" --yes
```

This runs cluster phase and

1. assets phase are ignored
2. network phase is validated and infrastructure will be created
3. IAM phase should only be a warning (not sure if this is fully working, as I mentioned in the PR above)
4. Security groups phase will be validated and updated warned if changes exist
5. Cluster phase will be synced

So that is where we are at, not certain what is in scope with this PR that will give you:

> If we get that right, we don't need #4434 for every single special-case.

I think we will need #4434 because phase does not exist unless I am not understanding how you want this done.  The IAM phase does not exist, so we cannot adjust how the lifecycle behaves. Please help me out here to understand if there is a better way.

Not needing phases for every different task sounds to me like a change in how the phases and lifecycles work with the tasks, which is kinda a lot more than what I am trying to do in this PR.  I like the idea, but is it simple to implement?  How do we implement it?

Need some more actionable direction here, please.

Reply:
Yes, I think we will still need #4434, cause we gotta have a phase. I can do the flag like that, but to me two flags are more readable.